xtevent/xtevent.ado

```diff
@@ -348,6 +374,67 @@ program define xtevent, eclass
 
 end
 
+* Program to parse window 
+program define parsewindow, rclass
+
+	syntax [anything] 
+		
+	tokenize "`anything'"
+	loc nwwindow = wordcount("`anything'")
+	if !inlist(`nwwindow',1,2) {
+		di as err _n "{bf:window} can only have one or two elements."
+		exit 198
+	}
+	
+	*check that all words are numeric or string 
+	loc isnum = 0
+	forvalues i=1/`nwwindow'{
+		cap confirm number ``i''
+		if !_rc loc ++ isnum 
+	}
+	if `isnum'>0 & `isnum'<`nwwindow'{
+		di as err _n "Invalid {bf:window} option."
+		exit 198
+	}
+	
+	* tell if all words are numeric or strings 
+	if `isnum' == 0 loc w_type ="string"
+	if `isnum' == `nwwindow' loc w_type ="numeric"
+	
+	* if all words are numbers, check that they are integers 
+	if "`w_type'"=="numeric" {
+		loc isnotint = 0
+		forvalues i=1/`nwwindow'{
+			cap confirm integer number ``i''
+			if _rc!=0 loc ++ isnotint 
+		}
+		if `isnotint'>0 {
+			di as err _n "Number in {bf:window} must be integer."
+			exit 126
+		}
+	}
+	
+	* if all words are string, check that it is only one word and it is a valid option name 
+	if "`w_type'"=="string" {
+		
+		if `nwwindow'>1 {
+			di as err _n "If string, {bf:window} must have only one element."
+			exit 198
+		}
+		
+		if `nwwindow'==1 {
+			if !inlist("`anything'","max","balanced"){
+				di as err _n "{bf:window} must be one of {bf:max} or {bf:balanced}."

```

Comment:
@Constantino-Carreto-Romero  "must be max or balanced"

Reply:
@jorpppp 
changed in https://github.com/JMSLab/xtevent/commit/4ee9081ff734c9efaaaaadc86b0033cee895ae22